app/models/case_contact.rb

```diff
@@ -21,9 +21,9 @@ class CaseContact < ApplicationRecord
     message: :cant_be_future,
     allow_nil: true
   }
-  validate :reimbursement_only_when_miles_driven, if: :active_or_expenses?
-  validate :volunteer_address_when_reimbursement_wanted, if: :active_or_expenses?
-  validate :volunteer_address_is_valid, if: :active_or_expenses?
+  validate :reimbursement_only_when_miles_driven, if: :active_or_details?
+  validate :volunteer_address_when_reimbursement_wanted, if: :active_or_details?
+  validate :volunteer_address_is_valid, if: :active_or_details?
```

Comment:
so it goes like:
- case_contacts/new => new record is `status: :started`
- form autosave does not save status during update (within format.json branch)
- html form submit (form_controller#update) sets `status: :details` prior to update (therefore uses the `:active_or_details?` validation)
- if update successful, `#finish_editing` sets `status: :active`
- if update fails, database case_contact will still be `status: :started`

Could just get rid of details status altogether and have contact in either `started` or `active` states. I think it would involve changing these validation ifs and changing the erb template name from `details.html.erb` to `started.html.erb`.  But I didn't want to deal with even more side effects on this PR, I think another ticket for that would be great.

Reply:
Gotcha that makes sense.